### PR TITLE
Feature/butter cms

### PR DESCRIPTION
**Pull request by mm716783**
Blog implementation, styles, prerendering, and webhooks

### CODE REVIEW

File: scripts/get-butter.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const fs = require('fs');
+
+const request = require('sync-request');
+const res = request('GET', 
+    'https://api.buttercms.com/v2/posts/?page=1&page_size=1000000&auth_token=fca1bc914c05371263f98c9d3480eaa215fd63d6',
```
Comment (Fdom92) [inline scripts/get-butter.js]: Is there a possibility to hide the auth token since this repo is public?

File: src/components/app-img/app-img.tsx [hunk 2]
```diff
@@ -57,7 +60,7 @@ export class Img {
   }
 
   private changeImageFormat() {
-    if (this.loadSrc && localStorage && localStorage.getItem('allowWebp') === 'true') {
+    if (this.loadSrc && 1 > 2 && (this.isServer || localStorage.getItem('allowWebp') === 'true')) {
```
Comment (Fdom92) [inline src/components/app-img/app-img.tsx]: why is this 1 > 2 thing? To ensure some conditions?

File: scripts/get-butter.js (hunk 1 above)
Comment (mm716783) [reply scripts/get-butter.js]: Hm, yes we could. The token will end up in the client though anyway (it's considered a client-safe token, plus the API is read-only, so basically no security risk), but I agree this is a good practice. 

Where's a good place to move it? Not sure if Stencil has an environment capability like Angular?

File: src/components/app-img/app-img.tsx (hunk 2 above)
Comment (mm716783) [reply src/components/app-img/app-img.tsx]: Ah, this was leftover from when I was testing something. Good catch, I will fix.

Comment (mm716783) [reply src/components/app-img/app-img.tsx]: Fixed

**Comment by mm716783** [comment]
Build works locally but Travis is failing...anybody have any ideas why?

**Comment by mm716783** [comment]
> Build works locally but Travis is failing...anybody have any ideas why?

Ok deleting package-lock.json resolved the Travis issue

File: scripts/get-butter.js (hunk 1 above)
Comment (Fdom92) [reply scripts/get-butter.js]: We can use something like in this post https://medium.com/stencil-tricks/environment-variables-with-stenciljs-57e9da591280

**Comment by Geoff-Openforge** [comment]
Things that need some styling changes:
1. Add spacing in between each post on the blog landing pages where the blogs are listed out and a horizontal line in between each blog as well. See: [](https://blog.ionicframework.com/)
2. The blog featured images stretch & distort when changing the browser window size
3. Can we make the blog featured images clickable to link to the blogs?
4. On the main featured post at the top of the landing page, can we make the title and image both link the to that blog?
5. Underline the page# that is active at the bottom of the blog landing page
6. When there aren't enough blogs to make the landing pages paginated, the bottom blog is right against the footer of the site. Can we add a fair amount of padding here?

**Comment by luiskcs89** [comment]
Fixed 1 and it should also fix 6 because it gives more padding.

File: scripts/get-butter.js (hunk 1 above)
Comment (mm716783) [reply scripts/get-butter.js]: This is now resolved with the merged branch from Sam (butter-api-key)

**Comment by Wholloran** [comment]
The following are items that came from the QA Pass:

1. The Profile image for "Billy" is displaying as broken on blogs with the publisher set to "billy" (As well as for Jedi). Note(For Same, Matt, and the receipts account displays a placeholder image from butter CMS, this is fine for users who don't have a profile picture on the website)
[https://openforge.teamwork.com/#tasks/13789798](url)

2. For filters with no content, this is being displayed: "No posts found for filter development." Let's change it to be "We don't currently have any blogs for this category. Check back soon!" This can be applied for all empty filter types.
[https://openforge.teamwork.com/#tasks/13789811](url)

3. The Read Next Section at the bottom of the blogs is blank with only a spinner. I let this sit for about 5 minutes and nothing populated. I'm assuming this should show the next three blog posts
[https://openforge.teamwork.com/#tasks/13789821](url)

4. The share buttons on the Blog pages are displaying as broken images
[https://openforge.teamwork.com/#tasks/13789828](url)

5. About the author, isn't populating the bio description for Billy & Jedi. 
[https://openforge.teamwork.com/#tasks/13789863](url)

6. /blog Margins: (Desktop) The blogs under the categories are not using the correct margins. It should be on the same margins as the featured blog post
[https://openforge.teamwork.com/#tasks/13789878](url)

7. /blog-post Margins (Desktop): The specific blog posts aren't using the correct margins, they should be aligned with the navbar margins
[https://openforge.teamwork.com/#tasks/13789885](url)

8. Mobile(): Button on the featured article says "READ MORE" But as per designs should say "See WHAT'S INSIDE"
[https://openforge.teamwork.com/#tasks/13789888](url)

9. Mobile /blog The user image should be smaller, it's causing the user's name to fall onto the second line even when they have shorter names 
[https://openforge.teamwork.com/#tasks/13789892](url)

10. mobile: when the drop-down is set to for announcements the line on the side of is will become shorter, however, when you switch back to a smaller category name ("Design") the lines don't go back to the correct size. attempt to mimic the mobile designs and we make any adjustments after the restructure is pushed)
[https://openforge.teamwork.com/#tasks/13789920](url)

11. Tablet(): /blog 
So we didn't have designs for a tablet but I talked with Rachel and we are going to have tablet have the stacked look (Like we do with mobile) in order to cut down on the strech images 
[https://openforge.teamwork.com/#tasks/13789929](url)

12. Tablet: bio-image on blog post page is being stretched
[https://openforge.teamwork.com/#tasks/13789939](url)

**Comment by PaulPaulDevelops** [comment]
#1, #4, and #5 are set within the ButterCMS portal. If you do not have social media, bio, or photo listed it will not display (as it is an empty source).

The others I will look into in the meantime. I wanted to make a note about those as there will not be fixes at the moment.

**Comment by Wholloran** [comment]
#3 is still happening

#11 looks way better overall, just going to make some more small requests
- While emulating on the iPad, the categories aren't following the correct Margin (they go out to the ends of the screen)

#12 is still occurring (Not it's occurring on the iPad not iPad pro)

**Pull request merged**